### ISSUE TITLE
`prep` won't exit in pushkin-cli (for Windows)

### ISSUE THREAD

**Issue by hzeng2** [pushkin-consortium/pushkin#208]
I was testing in the context of pushkin-cli and trying to get the experiment running and view it in a browser. I followed the steps laid out [here](https://github.com/pushkin-consortium/pushkin-sitetemplate-basic/pull/1#issuecomment-1590316874). 

When I ran `pushkin armageddon` and `node ../pushkin-cli/build/index.js prep`, it kept printing out `...` forever. And the output was: 

> 11:42 $ node ../pushkin-cli/build/index.js prep
Removing any .DS_Store files, if present.
package manager:  yarn
resetting experiments.js
Started prepping API for test_615
Started loading API controller for test_615
Using build-if-changed for  test_615_api
Installing dependencies for /home/haozeng/temp/experiments/test_615/api controllers
Started prepping web page for test_615
Started loading web page for test_615
Using build-if-changed for  test_615_web
Installing dependencies for /home/haozeng/temp/experiments/test_615/web page
loaded docker-compose.dev.yml and pushkin.yaml.
Building worker for test_615
Building docker image for test_615_worker
userMigsDir: /home/haozeng/temp/users/migrations
userDatabase: localtestdb
Loading migrations for test_615
expConfig:
 {"experimentName":"test 6/15","shortName":"test_615","apiControllers":{"mountPath":"test_615","location":"api controllers","name":"mycontroller"},"worker":{"location":"worker","service":{"image":"test_615","links":["message-queue","test_db"],"environment":{"AMQP_ADDRESS":"amqp://message-queue:5672","DB_USER":"postgres","DB_PASS":"example","DB_URL":"test_db","DB_NAME":"test_db"}}},"webPage":{"location":"web page"},"migrations":{"location":"migrations"},"seeds":{"location":""},"database":"localtestdb","logo":"logo512.png","text":"Enter your experiment description here.","tagline":"Be a citizen scientist! Try this quiz.","duration":"","productionDB":"Main"}
Waiting for database to start...
...
Building test_615_api from /home/haozeng/temp/experiments/test_615/api controllers
Building test_615_web from /home/haozeng/temp/experiments/test_615/web page
test_615_api is built
test_615_web is built
test_615_api is published locally via yalc
test_615_web is published locally via yalc
Loaded web page for test 6/15 (test_615_web)
Added test 6/15 to experiments.js
Writing out experiments.js
Installing combined API
Installed combined API
Installing combined front-end
Installed combined front-end
...
Finished building all experiment workers
updating docker-compose.dev.yml
Writing out front-end config
...
...
...
...
...
...
...
...
...
...
...
...
...
...
...

P.S. `node ../pushkin-cli/build/index.js start` did let me view the experiment in a browser. And the output was:

> 12:09 $ node ../pushkin-cli/build/index.js start
starting start...
Setting front-end 'environment variable'
running setEnv()
Setting front-end 'environment variable'
...
Pulling message-queue (rabbitmq:3.6-management)...
3.6-management: Pulling from library/rabbitmq
Digest: sha256:05bd722c6b0c747ae7f854baa9e9ceb9b47da75e39f93ad6a1db74f788b0f828
Status: Downloaded newer image for rabbitmq:3.6-management
Building api
Step 1/9 : FROM node:20.2
20.2: Pulling from library/node
Digest: sha256:14f0471d0478fbb9177d0f9e8c146dc872273dcdcfc7fea93a27ed81fc6b0e96
Status: Downloaded newer image for node:20.2
 ---> f03de6896e9e
Step 2/9 : COPY ./src /usr/src/app/src
 ---> a6fdf31344dc
Step 3/9 : COPY .yalc /usr/src/app/.yalc/
 ---> 7f5829618c57
Step 4/9 : COPY ./package.json ./yarn.lock ./.babelrc ./yalc.lock ./dockerStart.sh /usr/src/app/
 ---> bf4d04f6950a
Step 5/9 : WORKDIR /usr/src/app
 ---> Running in 50ec5fc041b0
Removing intermediate container 50ec5fc041b0
 ---> ccd0fab14078
Step 6/9 : RUN yarn install
 ---> Running in 814e6746e0ed
yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Done in 5.37s.
Removing intermediate container 814e6746e0ed
 ---> 2ed24436abb2
Step 7/9 : RUN yarn build
 ---> Running in 65aa77274131
yarn run v1.22.19
$ rm -rf build/* && babel src -d build && cp src/controllers.json build/
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
Successfully compiled 1 file with Babel (243ms).
Done in 0.45s.
Removing intermediate container 65aa77274131
 ---> a164c88b2388
Step 8/9 : RUN apt-get update && apt-get install -y netcat
 ---> Running in fef613bccb77
Get:1 http://deb.debian.org/debian bullseye InRelease [116 kB]
Get:2 http://deb.debian.org/debian-security bullseye-security InRelease [48.4 kB]
Get:3 http://deb.debian.org/debian bullseye-updates InRelease [44.1 kB]
Get:4 http://deb.debian.org/debian bullseye/main amd64 Packages [8183 kB]
Get:5 http://deb.debian.org/debian-security bullseye-security/main amd64 Packages [245 kB]
Get:6 http://deb.debian.org/debian bullseye-updates/main amd64 Packages [14.8 kB]
Fetched 8651 kB in 1s (7229 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  netcat-openbsd
The following NEW packages will be installed:
  netcat netcat-openbsd
0 upgraded, 2 newly installed, 0 to remove and 3 not upgraded.
Need to get 50.8 kB of archives.
After this operation, 130 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian bullseye/main amd64 netcat-openbsd amd64 1.217-3 [41.1 kB]
Get:2 http://deb.debian.org/debian bullseye/main amd64 netcat all 1.10-46 [9728 B]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 50.8 kB in 0s (1744 kB/s)
Selecting previously unselected package netcat-openbsd.
(Reading database ... 22794 files and directories currently installed.)
Preparing to unpack .../netcat-openbsd_1.217-3_amd64.deb ...
Unpacking netcat-openbsd (1.217-3) ...
Selecting previously unselected package netcat.
Preparing to unpack .../netcat_1.10-46_all.deb ...
Unpacking netcat (1.10-46) ...
Setting up netcat-openbsd (1.217-3) ...
update-alternatives: using /bin/nc.openbsd to provide /bin/nc (nc) in auto mode
Setting up netcat (1.10-46) ...
Removing intermediate container fef613bccb77
 ---> 0217e111eae0
Step 9/9 : CMD ["bash", "dockerStart.sh"]
 ---> Running in ac88e734432d
Removing intermediate container ac88e734432d
 ---> be9bd411b8df
Successfully built be9bd411b8df
Successfully tagged pushkin_api:latest
Building server
Step 1/10 : FROM node:20.2
 ---> f03de6896e9e
Step 2/10 : COPY ./public /usr/src/app/public
 ---> 634fe53f0822
Step 3/10 : COPY ./src /usr/src/app/src
 ---> 82ce21007132
Step 4/10 : COPY .yalc /usr/src/app/.yalc/
 ---> b6860270cc50
Step 5/10 : COPY ./experiments.js /usr/src/app/src/
 ---> ac3a00574fc3
Step 6/10 : COPY ./package.json ./yarn.lock ./yalc.lock ./craco.config.js /usr/src/app/
 ---> 70982022288a
Step 7/10 : WORKDIR /usr/src/app
 ---> Running in b21cc5628d39
Removing intermediate container b21cc5628d39
 ---> 4eb0642cf8c6
Step 8/10 : RUN yarn install
 ---> Running in b9f55871f69b
yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
warning "@craco/craco > autoprefixer@10.4.14" has unmet peer dependency "postcss@^8.1.0".
warning "@craco/craco > cosmiconfig-typescript-loader@1.0.9" has unmet peer dependency "@types/node@*".
warning "@craco/craco > cosmiconfig-typescript-loader > ts-node@10.9.1" has unmet peer dependency "@types/node@*".
warning " > bootstrap@4.5.0" has unmet peer dependency "jquery@1.9.1 - 3".
warning " > bootstrap@4.5.0" has unmet peer dependency "popper.js@^1.16.0".
warning "react-bootstrap > @restart/hooks@0.3.25" has incorrect peer dependency "react@^16.8.0".
warning " > react-redux@7.2.1" has incorrect peer dependency "react@^16.8.3".
warning "react-router-dom > react-router > mini-create-react-context@0.4.0" has incorrect peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0".
warning " > react-social-icons@4.1.0" has incorrect peer dependency "react@15.x.x || 16.x.x".
warning " > redux-form@8.3.6" has incorrect peer dependency "react@^16.4.2".
warning " > styled-components@5.1.1" has unmet peer dependency "react-is@>= 16.8.0".
warning " > @babel/plugin-syntax-flow@7.22.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning " > @babel/plugin-transform-react-jsx@7.22.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/plugin-transform-react-jsx > @babel/plugin-syntax-jsx@7.22.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning " > node-polyfill-webpack-plugin@2.0.1" has unmet peer dependency "webpack@>=5".
[4/4] Building fresh packages...
Done in 38.66s.
Removing intermediate container b9f55871f69b
 ---> b02eabdb8d2a
Step 9/10 : RUN apt-get update && apt-get install -y netcat
 ---> Running in 16d24c857d39
Get:1 http://deb.debian.org/debian bullseye InRelease [116 kB]
Get:2 http://deb.debian.org/debian-security bullseye-security InRelease [48.4 kB]
Get:3 http://deb.debian.org/debian bullseye-updates InRelease [44.1 kB]
Get:4 http://deb.debian.org/debian bullseye/main amd64 Packages [8183 kB]
Get:5 http://deb.debian.org/debian-security bullseye-security/main amd64 Packages [245 kB]
Get:6 http://deb.debian.org/debian bullseye-updates/main amd64 Packages [14.8 kB]
Fetched 8651 kB in 1s (7507 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  netcat-openbsd
The following NEW packages will be installed:
  netcat netcat-openbsd
0 upgraded, 2 newly installed, 0 to remove and 3 not upgraded.
Need to get 50.8 kB of archives.
After this operation, 130 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian bullseye/main amd64 netcat-openbsd amd64 1.217-3 [41.1 kB]
Get:2 http://deb.debian.org/debian bullseye/main amd64 netcat all 1.10-46 [9728 B]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 50.8 kB in 0s (1235 kB/s)
Selecting previously unselected package netcat-openbsd.
(Reading database ... 22794 files and directories currently installed.)
Preparing to unpack .../netcat-openbsd_1.217-3_amd64.deb ...
Unpacking netcat-openbsd (1.217-3) ...
Selecting previously unselected package netcat.
Preparing to unpack .../netcat_1.10-46_all.deb ...
Unpacking netcat (1.10-46) ...
Setting up netcat-openbsd (1.217-3) ...
update-alternatives: using /bin/nc.openbsd to provide /bin/nc (nc) in auto mode
Setting up netcat (1.10-46) ...
Removing intermediate container 16d24c857d39
 ---> c4ed639dd0f7
Step 10/10 : CMD yarn start
 ---> Running in 27678adff339
Removing intermediate container 27678adff339
 ---> cda69bfd262b
Successfully built cda69bfd262b
Successfully tagged pushkin_server:latest
pushkin_test_db_1 is up-to-date
Creating pushkin_message-queue_1 ... done
Creating pushkin_api_1             ... done
Creating pushkin_test_615_worker_1 ... done                                                                                           Creating pushkin_server_1          ... done
3.6-management: Pulling from library/rabbitmq
Digest: sha256:05bd722c6b0c747ae7f854baa9e9ceb9b47da75e39f93ad6a1db74f788b0f828
Status: Downloaded newer image for rabbitmq:3.6-management
Step 1/9 : FROM node:20.2
20.2: Pulling from library/node
Digest: sha256:14f0471d0478fbb9177d0f9e8c146dc872273dcdcfc7fea93a27ed81fc6b0e96
Status: Downloaded newer image for node:20.2
 ---> f03de6896e9e
Step 2/9 : COPY ./src /usr/src/app/src
 ---> a6fdf31344dc
Step 3/9 : COPY .yalc /usr/src/app/.yalc/
 ---> 7f5829618c57
Step 4/9 : COPY ./package.json ./yarn.lock ./.babelrc ./yalc.lock ./dockerStart.sh /usr/src/app/
 ---> bf4d04f6950a
Step 5/9 : WORKDIR /usr/src/app
 ---> Running in 50ec5fc041b0
Removing intermediate container 50ec5fc041b0
 ---> ccd0fab14078
Step 6/9 : RUN yarn install
 ---> Running in 814e6746e0ed
yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
Done in 5.37s.
Removing intermediate container 814e6746e0ed
 ---> 2ed24436abb2
Step 7/9 : RUN yarn build
 ---> Running in 65aa77274131
yarn run v1.22.19
$ rm -rf build/* && babel src -d build && cp src/controllers.json build/
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
Successfully compiled 1 file with Babel (243ms).
Done in 0.45s.
Removing intermediate container 65aa77274131
 ---> a164c88b2388
Step 8/9 : RUN apt-get update && apt-get install -y netcat
 ---> Running in fef613bccb77
Get:1 http://deb.debian.org/debian bullseye InRelease [116 kB]
Get:2 http://deb.debian.org/debian-security bullseye-security InRelease [48.4 kB]
Get:3 http://deb.debian.org/debian bullseye-updates InRelease [44.1 kB]
Get:4 http://deb.debian.org/debian bullseye/main amd64 Packages [8183 kB]
Get:5 http://deb.debian.org/debian-security bullseye-security/main amd64 Packages [245 kB]
Get:6 http://deb.debian.org/debian bullseye-updates/main amd64 Packages [14.8 kB]
Fetched 8651 kB in 1s (7229 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  netcat-openbsd
The following NEW packages will be installed:
  netcat netcat-openbsd
0 upgraded, 2 newly installed, 0 to remove and 3 not upgraded.
Need to get 50.8 kB of archives.
After this operation, 130 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian bullseye/main amd64 netcat-openbsd amd64 1.217-3 [41.1 kB]
Get:2 http://deb.debian.org/debian bullseye/main amd64 netcat all 1.10-46 [9728 B]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 50.8 kB in 0s (1744 kB/s)
Selecting previously unselected package netcat-openbsd.
(Reading database ... 22794 files and directories currently installed.)
Preparing to unpack .../netcat-openbsd_1.217-3_amd64.deb ...
Unpacking netcat-openbsd (1.217-3) ...
Selecting previously unselected package netcat.
Preparing to unpack .../netcat_1.10-46_all.deb ...
Unpacking netcat (1.10-46) ...
Setting up netcat-openbsd (1.217-3) ...
update-alternatives: using /bin/nc.openbsd to provide /bin/nc (nc) in auto mode
Setting up netcat (1.10-46) ...
Removing intermediate container fef613bccb77
 ---> 0217e111eae0
Step 9/9 : CMD ["bash", "dockerStart.sh"]
 ---> Running in ac88e734432d
Removing intermediate container ac88e734432d
 ---> be9bd411b8df
Successfully built be9bd411b8df
Successfully tagged pushkin_api:latest
Step 1/10 : FROM node:20.2
 ---> f03de6896e9e
Step 2/10 : COPY ./public /usr/src/app/public
 ---> 634fe53f0822
Step 3/10 : COPY ./src /usr/src/app/src
 ---> 82ce21007132
Step 4/10 : COPY .yalc /usr/src/app/.yalc/
 ---> b6860270cc50
Step 5/10 : COPY ./experiments.js /usr/src/app/src/
 ---> ac3a00574fc3
Step 6/10 : COPY ./package.json ./yarn.lock ./yalc.lock ./craco.config.js /usr/src/app/
 ---> 70982022288a
Step 7/10 : WORKDIR /usr/src/app
 ---> Running in b21cc5628d39
Removing intermediate container b21cc5628d39
 ---> 4eb0642cf8c6
Step 8/10 : RUN yarn install
 ---> Running in b9f55871f69b
yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
warning "@craco/craco > autoprefixer@10.4.14" has unmet peer dependency "postcss@^8.1.0".
warning "@craco/craco > cosmiconfig-typescript-loader@1.0.9" has unmet peer dependency "@types/node@*".
warning "@craco/craco > cosmiconfig-typescript-loader > ts-node@10.9.1" has unmet peer dependency "@types/node@*".
warning " > bootstrap@4.5.0" has unmet peer dependency "jquery@1.9.1 - 3".
warning " > bootstrap@4.5.0" has unmet peer dependency "popper.js@^1.16.0".
warning "react-bootstrap > @restart/hooks@0.3.25" has incorrect peer dependency "react@^16.8.0".
warning " > react-redux@7.2.1" has incorrect peer dependency "react@^16.8.3".
warning "react-router-dom > react-router > mini-create-react-context@0.4.0" has incorrect peer dependency "react@^0.14.0 || ^15.0.0 || ^16.0.0".
warning " > react-social-icons@4.1.0" has incorrect peer dependency "react@15.x.x || 16.x.x".
warning " > redux-form@8.3.6" has incorrect peer dependency "react@^16.4.2".
warning " > styled-components@5.1.1" has unmet peer dependency "react-is@>= 16.8.0".
warning " > @babel/plugin-syntax-flow@7.22.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning " > @babel/plugin-transform-react-jsx@7.22.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@babel/plugin-transform-react-jsx > @babel/plugin-syntax-jsx@7.22.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning " > node-polyfill-webpack-plugin@2.0.1" has unmet peer dependency "webpack@>=5".
[4/4] Building fresh packages...
Done in 38.66s.
Removing intermediate container b9f55871f69b
 ---> b02eabdb8d2a
Step 9/10 : RUN apt-get update && apt-get install -y netcat
 ---> Running in 16d24c857d39
Get:1 http://deb.debian.org/debian bullseye InRelease [116 kB]
Get:2 http://deb.debian.org/debian-security bullseye-security InRelease [48.4 kB]
Get:3 http://deb.debian.org/debian bullseye-updates InRelease [44.1 kB]
Get:4 http://deb.debian.org/debian bullseye/main amd64 Packages [8183 kB]
Get:5 http://deb.debian.org/debian-security bullseye-security/main amd64 Packages [245 kB]
Get:6 http://deb.debian.org/debian bullseye-updates/main amd64 Packages [14.8 kB]
Fetched 8651 kB in 1s (7507 kB/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
The following additional packages will be installed:
  netcat-openbsd
The following NEW packages will be installed:
  netcat netcat-openbsd
0 upgraded, 2 newly installed, 0 to remove and 3 not upgraded.
Need to get 50.8 kB of archives.
After this operation, 130 kB of additional disk space will be used.
Get:1 http://deb.debian.org/debian bullseye/main amd64 netcat-openbsd amd64 1.217-3 [41.1 kB]
Get:2 http://deb.debian.org/debian bullseye/main amd64 netcat all 1.10-46 [9728 B]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 50.8 kB in 0s (1235 kB/s)
Selecting previously unselected package netcat-openbsd.
(Reading database ... 22794 files and directories currently installed.)
Preparing to unpack .../netcat-openbsd_1.217-3_amd64.deb ...
Unpacking netcat-openbsd (1.217-3) ...
Selecting previously unselected package netcat.
Preparing to unpack .../netcat_1.10-46_all.deb ...
Unpacking netcat (1.10-46) ...
Setting up netcat-openbsd (1.217-3) ...
update-alternatives: using /bin/nc.openbsd to provide /bin/nc (nc) in auto mode
Setting up netcat (1.10-46) ...
Removing intermediate container 16d24c857d39
 ---> c4ed639dd0f7
Step 10/10 : CMD yarn start
 ---> Running in 27678adff339
Removing intermediate container 27678adff339
 ---> cda69bfd262b
Successfully built cda69bfd262b
Successfully tagged pushkin_server:latest
 Starting. You may not be able to load localhost for a minute or two.

**Comment by jkhartshorne** [comment]
Your database isn't starting during `prep`. Try running it agin. Once it gets stuck in the `...` loop, please open Docker Desktop and look at what containers are available. Please include a screenshot. If possible, open a terminal in Docker Desktop to the container with the database and copy the log and paste here.

**Comment by hzeng2** [comment]
Please see the log:

 

> 17:54 $ sudo docker container logs --details 22dbcf7dd4a6
>  The files belonging to this database system will be owned by user "postgres".
>  This user must also own the server process.
> 
>  The database cluster will be initialized with locale "en_US.utf8".
>  The default database encoding has accordingly been set to "UTF8".
>  The default text search configuration will be set to "english".
> 
>  Data page checksums are disabled.
> 
>  fixing permissions on existing directory /var/lib/postgresql/data ... ok
>  creating subdirectories ... ok
>  selecting default max_connections ... 100
>  selecting default shared_buffers ... 128MB
>  selecting default timezone ... Etc/UTC
>  selecting dynamic shared memory implementation ... posix
>  creating configuration files ... ok
>  running bootstrap script ... ok
>  performing post-bootstrap initialization ... ok
>  syncing data to disk ... ok
> 
>  Success. You can now start the database server using:
> 
>      pg_ctl -D /var/lib/postgresql/data -l logfile start
> 
> 
>  WARNING: enabling "trust" authentication for local connections
>  You can change this by editing pg_hba.conf or using the option -A, or
>  --auth-local and --auth-host, the next time you run initdb.
>  waiting for server to start....2023-06-16 00:51:38.493 UTC [49] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
>  2023-06-16 00:51:38.528 UTC [50] LOG:  database system was shut down at 2023-06-16 00:51:37 UTC
>  2023-06-16 00:51:38.536 UTC [49] LOG:  database system is ready to accept connections
>   done
>  server started
>  CREATE DATABASE
> 
> 
>  /usr/local/bin/docker-entrypoint.sh: ignoring /docker-entrypoint-initdb.d/*
> 
>  waiting for server to shut down....2023-06-16 00:51:38.909 UTC [49] LOG:  received fast shutdown request
>  2023-06-16 00:51:38.917 UTC [49] LOG:  aborting any active transactions
>  2023-06-16 00:51:38.920 UTC [49] LOG:  background worker "logical replication launcher" (PID 56) exited with exit code 1
>  2023-06-16 00:51:38.920 UTC [51] LOG:  shutting down
>  2023-06-16 00:51:38.971 UTC [49] LOG:  database system is shut down
>   done
>  server stopped
> 
>  PostgreSQL init process complete; ready for start up.
> 
>  2023-06-16 00:51:39.023 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
>  2023-06-16 00:51:39.024 UTC [1] LOG:  listening on IPv6 address "::", port 5432
>  2023-06-16 00:51:39.038 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
>  2023-06-16 00:51:39.068 UTC [77] LOG:  database system was shut down at 2023-06-16 00:51:38 UTC
>  2023-06-16 00:51:39.078 UTC [1] LOG:  database system is ready to accept connections

**Comment by jkhartshorne** [comment]
So it looks like the database is running, but for some reason it isn't accepting connections. Here is what you should see for `pushkin prep`:

> jkhartshorne@MacBook-Air-345 temp % node ../pushkin-cli/build/index.js prep      
Removing any .DS_Store files, if present.
/Users/jkhartshorne/Documents/Projects/PushkinDev/PushkinNew/temp/experiments
/Users/jkhartshorne/Documents/Projects/PushkinDev/PushkinNew/temp/pushkin
package manager:  yarn
resetting experiments.js
Started prepping API for yoyoyo
Started loading API controller for yoyoyo
Using build-if-changed for  yoyoyo_api
Installing dependencies for /Users/jkhartshorne/Documents/Projects/PushkinDev/PushkinNew/temp/experiments/yoyoyo/api controllers
Started prepping web page for yoyoyo
Started loading web page for yoyoyo
Using build-if-changed for  yoyoyo_web
Installing dependencies for /Users/jkhartshorne/Documents/Projects/PushkinDev/PushkinNew/temp/experiments/yoyoyo/web page
loaded docker-compose.dev.yml and pushkin.yaml.
Building worker for yoyoyo
Building docker image for yoyoyo_worker
userMigsDir: /Users/jkhartshorne/Documents/Projects/PushkinDev/PushkinNew/temp/users/migrations
userDatabase: localtestdb
Loading migrations for yoyoyo
expConfig:
 {"experimentName":"yoyoyo","shortName":"yoyoyo","apiControllers":{"mountPath":"yoyoyo","location":"api controllers","name":"mycontroller"},"worker":{"location":"worker","service":{"image":"yoyoyo","links":["message-queue","test_db"],"environment":{"AMQP_ADDRESS":"amqp://message-queue:5672","DB_USER":"postgres","DB_PASS":"example","DB_URL":"test_db","DB_NAME":"test_db"}}},"webPage":{"location":"web page"},"migrations":{"location":"migrations"},"seeds":{"location":""},"database":"localtestdb","logo":"logo512.png","text":"Enter your experiment description here.","tagline":"Be a citizen scientist! Try this quiz.","duration":"","productionDB":"Main"}
Waiting for database to start...
...
Building yoyoyo_web from /Users/jkhartshorne/Documents/Projects/PushkinDev/PushkinNew/temp/experiments/yoyoyo/web page
Building yoyoyo_api from /Users/jkhartshorne/Documents/Projects/PushkinDev/PushkinNew/temp/experiments/yoyoyo/api controllers
yoyoyo_web is built
yoyoyo_api is built
yoyoyo_web is published locally via yalc
Loaded web page for yoyoyo (yoyoyo_web)
Added yoyoyo to experiments.js
Writing out experiments.js
Installing combined API
Installed combined API
Installing combined front-end
Installed combined front-end
yoyoyo_api is published locally via yalc
...
...
...
Finished building all experiment workers
updating docker-compose.dev.yml
Writing out front-end config
...
...
starting migrations....
migAndSeedDirs:
 [
  {
    migrations: '/Users/jkhartshorne/Documents/Projects/PushkinDev/PushkinNew/temp/users/migrations',
    seeds: ''
  },
  {
    migrations: '/Users/jkhartshorne/Documents/Projects/PushkinDev/PushkinNew/temp/experiments/yoyoyo/migrations',
    seeds: '/Users/jkhartshorne/Documents/Projects/PushkinDev/PushkinNew/temp/experiments/yoyoyo'
  }
]
db: localtestdb
FS-related option specified for migration configuration. This resets migrationSource to default FsMigrations
Ran migrations for main database
Running seeds on /Users/jkhartshorne/Documents/Projects/PushkinDev/PushkinNew/temp/experiments/yoyoyo
Set up databases successfully
 Database updated. Shutting down...

As you can see, the database migrations were never run when you ran `pushkin prep`. 

Without you being here with your computer, this is very hard to troubleshoot. You are probably going to have to figure this out on your own. A few things though:

1. I assume you'll be attaching the screenshot I asked for later. 
2. Also, please confirm which commit of the CLI you are using. 
3. After running `prep`, Use SQLPro for Postgres (or similar) or just use the terminal within the docker container to check for any tables in the database. 
4. Do the same thing after running `start`. Is the database actually working after you run `start`? I would doubt it. It would be helpful to get the full log from the container and well as the CLI output after running `start`.

**Comment by hzeng2** [comment]
I figured out a **temporary** fix! The reason why the database would not fire up during `prep` is because it still uses the old docker container name. I was checking the container and noticed the container name and recalled this [fix](https://github.com/pushkin-consortium/pushkin/issues/195#issuecomment-1523884247). I also made sure I was using the updated pushkin-cli/master branch all along. But somehow the container name was not updated automatically.

Then I tried manually changing the container name:
`docker rename pushkin_test_db_1 pushkin-test_db-1`

![image](https://github.com/pushkin-consortium/pushkin/assets/90575813/758915e6-aecd-4705-97e1-d4fb03798e48)
Now `prep` and `start` are both working properly for me. However, whenever I stop and restart docker / run `pushkin armageddon`, I have to change the container name again. Therefore this is only a **temporary** fix.

@jkhartshorne -- I am not sure whether this is a random hiccup pertained to my laptop or something that could get protected from the codebase though.

**Comment by jkhartshorne** [comment]
Yah, I'm not sure how that happened. The current version of the code definitely names it `pushkin-test_db-1`. 

Do me a favor and open `pushkin-cli/src/commands/setupdb/index.js` Then check to see if you can find the following code:

`  //By this point, with any luck the DB is up and running.
  //But just in case, this ridiculously roundabout loop...
  const wait = async () => {
    //Sometimes, I really miss loops
    let x = await exec (`docker ps -f name=pushkin-test_db-1`)
          .then((x) => {
            console.log('...')
            return x.stdout.search('healthy')
          },
          err => {console.error(err)})
    if (x > 0) {
`

See what the database is named in there. Then also check to see what commit you are using.

**Comment by hzeng2** [comment]
@jkhartshorne -- I can confirm that I am on the master branch of cli and have pulled the latest commit. Also, I am able to see the code for naming database in `pushkin-cli/src/commands/setupdb/index.js`:

![image](https://github.com/pushkin-consortium/pushkin/assets/90575813/caa845d1-e63c-4bc0-9a8e-4a1c3fca1cfc)

The issue might be with `pushkin armageddon`. This is how I found out: I was able to run `prep` with no problem today. No manual change of container name needed. However, after I ran `pushkin armageddon`, the container name went back to its outdated version. And that overwrites the naming of any following containers. I tried removing the container and then started a new one with a new ID. The new container's name didn't get updated.
